src/types.ts

```diff
@@ -10,10 +11,11 @@ export type SeedPhrase = string;
 
 export type NetworkReward = Winston;
 
-export type FolderID = string;
-export type FileID = string;
-export type DriveID = string;
-export type EntityID = DriveID | FolderID | FileID;
+export type RootFolderID = 'root folder';
+export type FolderID = EntityID | RootFolderID;
```

Comment:
I don't think this is technically correct here. And this is also causing the type issues downstream where you have to cast things `as EntityID` in a couple places in the PR.

A folder's `EntityID` will never be `'root folder'`. That string is only ever used as a place holder for the `parentFolderId` field when it is missing. This case should only happen on the root folder of a drive. So, `folderId` will never be root folder, only the `parentFolderId` field.

```suggestion
export type FolderID = EntityID;
```

Reply:
Yep those placeholder values were causing exceptions at runtime. I'm going to try the same pattern we used for EncryptedEntityId.

Reply:
2738016f1886bf5b426e20a42adfbad322765fd6